PreFreeSurfer/PreFreeSurferPipeline.sh

```diff
@@ -452,253 +488,311 @@ log_Msg "POSIXLY_CORRECT="${POSIXLY_CORRECT}
-		--fnirtconfig=${FNIRTConfig}
-
-done
-
-# End of looping over modalities (T1w and T2w)
+# --- skip if we have a custom brain
+
+if [ "$CustomBrain" = "NONE" ] ; then
+
+  Modalities="T1w T2w"
+
+  for TXw in ${Modalities} ; do
+
+      # skip modality if no image
+
+      if [ "${TXwInputImages}" = "NONE" ] ; then
+        log_Msg "Skipping Modality: " $TXw " - image not specified."
+        continue
+      else
+        log_Msg "Processing Modality: " $TXw
+      fi
+
+      # set up appropriate input variables
+      if [ $TXw = T1w ] ; then
+          TXwInputImages="${T1wInputImages}"
+          TXwFolder=${T1wFolder}
+          TXwImage=${T1wImage}
+          TXwTemplate=${T1wTemplate}
+          TXwTemplate2mm=${T1wTemplate2mm}
+      else
+          TXwInputImages="${T2wInputImages}"
+          TXwFolder=${T2wFolder}
+          TXwImage=${T2wImage}
+          TXwTemplate=${T2wTemplate}
+          TXwTemplate2mm=${T2wTemplate2mm}
+      fi
+      OutputTXwImageSTRING=""
+
+      # Perform Gradient Nonlinearity Correction
+
+      if [ ! $GradientDistortionCoeffs = "NONE" ] ; then
+        log_Msg "Performing Gradient Nonlinearity Correction"
+
+        i=1
+        for Image in $TXwInputImages ; do
+          wdir=${TXwFolder}/${TXwImage}${i}_GradientDistortionUnwarp
+          log_Msg "mkdir -p $wdir"
+          mkdir -p $wdir
+          # Make sure input axes are oriented the same as the templates
+          ${RUN} ${FSLDIR}/bin/fslreorient2std $Image ${wdir}/${TXwImage}${i}
+
+          ${RUN} ${HCPPIPEDIR_Global}/GradientDistortionUnwarp.sh \
+            --workingdir=${wdir} \
+            --coeffs=$GradientDistortionCoeffs \
+            --in=${wdir}/${TXwImage}${i} \
+            --out=${TXwFolder}/${TXwImage}${i}_gdc \
+            --owarp=${TXwFolder}/xfms/${TXwImage}${i}_gdc_warp
+          OutputTXwImageSTRING="${OutputTXwImageSTRING}${TXwFolder}/${TXwImage}${i}_gdc "
+          i=$(($i+1))
+        done
+
+      else
+        log_Msg "NOT PERFORMING GRADIENT DISTORTION CORRECTION"
+
+        i=1
+        for Image in $TXwInputImages ; do
+          ${RUN} ${FSLDIR}/bin/fslreorient2std $Image ${TXwFolder}/${TXwImage}${i}_gdc
+          OutputTXwImageSTRING="${OutputTXwImageSTRING}${TXwFolder}/${TXwImage}${i}_gdc "
+          i=$(($i+1))
+        done
+
+      fi
+
+      # Average Like (Same Modality) Scans
+
+      if [ `echo $TXwInputImages | wc -w` -gt 1 ] ; then
+        log_Msg "Averaging ${TXw} Images"
+        log_Msg "mkdir -p ${TXwFolder}/Average${TXw}Images"
+        mkdir -p ${TXwFolder}/Average${TXw}Images
+        log_Msg "PERFORMING SIMPLE AVERAGING"
+        ${RUN} ${HCPPIPEDIR_PreFS}/AnatomicalAverage.sh -o ${TXwFolder}/${TXwImage} -s ${TXwTemplate} -m ${TemplateMask} -n -w ${TXwFolder}/Average${TXw}Images --noclean -v -b $BrainSize $OutputTXwImageSTRING
+      else
+        log_Msg "Not Averaging ${TXw} Images"
+        log_Msg "ONLY ONE AVERAGE FOUND: COPYING"
```

Comment:
This probably should have said "image", not "average".

Reply:
Corrected.